app/src/androidTest/java/com/alexzh/testapp/ui/home/HomeActivityTest.kt

```diff
@@ -1,45 +1,165 @@
+     *
+     * Hint(s):
+     *  - the number of items in RecyclerView can be checked with RecyclerViewMatchers#withItemCount
+     *  ViewMatcher (can be not implemented yet).
+     */
     @Test
     fun shouldRecyclerViewHas15Items() {
         fail()
     }
 
+    /**
+     * Test case:
+     *  - HomeActivity should have options menu
+     *  - SettingsActivity should be opened after pressing on "Settings" in options menu
+     *
+     * Notes:
+     *  - the "Settings" text can be found in strings.xml file as settings_action const
+     *  (R.string.settings_action)
+     *
+     * Hint(s):
+     *  - Intent.intended(IntentMatcher) allows us to verify that Activity was opened.
+     *  The Intents component should be initialized before with the Intents#init() method
+     *  and released when it's not needed with the Intents#release() method.
+     *  - the Espresso#openActionBarOverflowOrOptionsMenu method allows us to open options menu.
+     */
     @Test
     fun shouldOpenSettingsScreenWhenSettingsItemClicked() {
         fail()
     }
 
+    /**
+     * Test case:
+     *  - the task title should be displayed after click on item from RecyclerView
+     *
+     * Notes:
+     *  - the "RecyclerView" has ID: recyclerView (R.id.recyclerView)
+     *  - the "RecyclerView" shows items created in DummyData#getTasks() method
+     *
+     * Hint(s):
+     *  - the action on item can be made with RecyclerViewActions#actionOnItem method.
+     */
     @Test
     fun shouldBeDisplayedTaskTitleWhenClickOnTask() {
         fail()
     }
 
+    /**
+     * Test case:
+     *  - the "The {task title} task is favourite" or "The {task title} task is not favourite" text
+     *  should be displayed in Snackbar after click on favourite icon of the item, depends on
+     *  the status of the item.
+     *
+     * Notes:
+     *  - the data displayed in the RecyclerView can found in DummyData#getTasks() method.
+     *  - the "The {task title} task is favourite" string can be found in string.xml file as
+     *  formatted_favourite_task const (R.string.formatted_favourite_task).
+     *  - the "The {task title} task is not favourite" string can be found in string.xml file as
+     *  formatted_not_favourite_task const (R.string.formatted_not_favourite_task).
+     *  - the Snackbar component displays text which can be found in view with ID: snackbar_text
+     *  (com.google.android.material.R.id.snackbar_text).
+     *
+     * Hint(s):
+     *  - the click on child item, like favourite icon can be made with
+     *  RecyclerViewActions#clickByChildViewWithId ViewAction (can be not implemented yet).
+     */
     @Test
     fun shouldBeDisplayTaskInfoWhenClickedOnFavouriteIcon() {
         fail()
     }
 
+    /**
+     * Test case:
+     *  - the first task should be favourite. We can verify it by verification ViewHolder of the
+     *  task. Each item has two descendants: title and icon.
```

Comment:
I think it can be helpful for I'mBarista application because this app is small and here we have only 7 test cases in the file. I want to have a similar structure as in real project and I want to split into files based on business logic.